File: frontend/src/app/api-paths.ts
const API_BASE = "/api";

export const API_PATHS = {
    auth: {
        base: `${API_BASE}/auth`,
        login: `${API_BASE}/auth/login`,
        signup: `${API_BASE}/auth/signup`,
        me: `${API_BASE}/auth/me`,
        logout: `${API_BASE}/auth/logout`,
    },
    events: {
        base: `${API_BASE}/events`,
        byId: (eventId: string) => `${API_BASE}/events/${eventId}`,
        attendeeEvents: `${API_BASE}/events/attendee-events`,
    },
    tickets: {
        base: `${API_BASE}/tickets`,
        purchase: `${API_BASE}/tickets/purchase`,
    },
} as const;

File: backend/src/main/java/com/ticketplatform/backend/config/ApiPaths.java
package com.ticketplatform.backend.config;

public final class ApiPaths {

    private ApiPaths() {
    }

    public static final String API = "/api";

    public static final class Auth {
        private Auth() {
        }

        public static final String BASE = API + "/auth";
        public static final String LOGIN = BASE + "/login";
        public static final String SIGNUP = BASE + "/signup";
        public static final String ME = BASE + "/me";
        public static final String LOGOUT = BASE + "/logout";
    }

    public static final class Events {
        private Events() {
        }

        public static final String BASE = API + "/events";
        public static final String ANY = BASE + "/**";
    }

    public static final class Tickets {
        private Tickets() {
        }

        public static final String BASE = API + "/tickets";
        public static final String ME = BASE + "/me";
        public static final String ANY = BASE + "/**";
    }

    public static final class Cors {
        private Cors() {
        }

        public static final String FRONTEND_LOCAL = "http://localhost:5173";
    }
}
